tests/vmi_configuration_test.go

```diff
@@ -2110,4 +2110,19 @@ var _ = Describe("Configurations", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
+
+	FIt("Masquerede VMI with Guest agent should not update VMI IP", func() {
+		vmi := tests.NewRandomFedoraVMIWitGuestAgent()
+
+		By("Starting a VirtualMachineInstance")
+		vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+		Expect(err).ToNot(HaveOccurred())
+		tests.WaitForSuccessfulVMIStart(vmi)
+
+		vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+		Expect(vmi.Status.Interfaces[0].IP).To(Equal(vmiPod.Status.PodIP))
```

Comment:
@vatsalparekh, @enp0s3 made me realize there's a race condition here. Ensuring that the Agent is running helps, but there's a still finite amount of time between when the agent reports as ready and when the IP gets updated. This Expect should probably be replaced with an Eventually clause.

Reply:
+1

Reply:
Added `Eventually`